File: src/main/java/com/googongill/aditory/controller/dto/user/response/UserTokenResponse.java
package com.googongill.aditory.controller.dto.user.response;

import com.fasterxml.jackson.annotation.JsonAutoDetect;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.googongill.aditory.service.dto.category.CategoryResult;
import com.googongill.aditory.service.dto.user.UserTokenResult;
import lombok.Builder;

import java.util.ArrayList;
import java.util.List;

@Builder
@JsonSerialize
@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
public class UserTokenResponse {
    private Long userId;
    private String username;
    private String nickname;
    private String contact;
    private String accessToken;
    private String refreshToken;
    @Builder.Default
    private List<CategoryResult> userCategories = new ArrayList<>();


    public static UserTokenResponse of(UserTokenResult userTokenResult) {
        return UserTokenResponse.builder()
                .userId(userTokenResult.getUserId())
                .username(userTokenResult.getUsername())
                .nickname(userTokenResult.getNickname())
                .contact(userTokenResult.getContact())
                .accessToken(userTokenResult.getAccessToken())
                .refreshToken(userTokenResult.getRefreshToken())
                .userCategories(userTokenResult.getUserCategories())
                .build();
    }
}

File: src/main/java/com/googongill/aditory/controller/dto/user/response/UserTokenResponse.kt
package com.googongill.aditory.controller.dto.user.response

import com.fasterxml.jackson.annotation.JsonAutoDetect
import com.fasterxml.jackson.databind.annotation.JsonSerialize
import com.googongill.aditory.service.dto.category.CategoryResult
import com.googongill.aditory.service.dto.user.UserTokenResult

@JsonSerialize
@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
data class UserTokenResponse(
    val userId: Long?,
    val username: String,
    val nickname: String?,
    val contact: String?,
    val accessToken: String?,
    val refreshToken: String?,
    val userCategories: List<CategoryResult> = emptyList()

) {

    companion object {
        fun of(userTokenResult: UserTokenResult): UserTokenResponse {
            return UserTokenResponse(
                userId = userTokenResult.userId,
                username = userTokenResult.username,
                nickname = userTokenResult.nickname,
                contact = userTokenResult.contact,
                accessToken = userTokenResult.accessToken,
                refreshToken = userTokenResult.refreshToken,
                userCategories = userTokenResult.userCategories
            )
        }
    }

}
